pkg/nodediscovery/nodediscovery.go

```diff
@@ -298,7 +299,11 @@ func (n *NodeDiscovery) UpdateCiliumNodeResource() {
 			}
 		}
 
-		n.mutateNodeResource(nodeResource)
+		err = n.mutateNodeResource(nodeResource)
+		if err != nil {
+			log.WithError(err).WithField("retryCount", retryCount).Warning("unable to mutate nodeResource")
+			continue
+		}
```

Comment:
We can inline the error here and capitalize the first word of a log msg to conform on style with the rest of the log msgs.

```suggestion
		if err ;= n.mutateNodeResource(nodeResource); err != nil {
			log.WithError(err).WithField("retryCount", retryCount).Warning("Unable to mutate nodeResource")
			continue
		}
```